### ISSUE TITLE
Catch-all Routes. Map multiple 404 routes.

### ISSUE THREAD

**Issue by TakenHiduk** [hannoeru/vite-plugin-pages#136]
Hello, I've a doubt, can I have multiple 404 pages? since I'm gonna have multiple "apps" in my page.

| Tecnologies | Version |
| ------------- | ------------- |
| Yarn | 1.22.17 |
| Node | 16.13.0 |
| NPM | 8.1.4 |

This is my package.json
```` 
{
  "name": "salepush",
  "version": "0.0.0",
  "engines": {
    "node": ">=14.13",
    "npm": ">=8.1.4"
  },
  "main": "server.js",
  "scripts": {
    "start": "node server.js",
    "dev": "vite",
    "build": "vite-ssr build",
    "preview": "vite preview",
    "lint": "eslint src/**/*.{ts,vue}",
    "format": "prettier --write src/**/*.{ts,vue}"
  },
  "dependencies": {
    "@fortawesome/fontawesome-svg-core": "^1.2.36",
    "@fortawesome/free-brands-svg-icons": "^5.15.4",
    "@fortawesome/free-regular-svg-icons": "^5.15.4",
    "@fortawesome/free-solid-svg-icons": "^5.15.4",
    "@fortawesome/vue-fontawesome": "^3.0.0-5",
    "@vue/server-renderer": "^3.2.26",
    "@vueuse/head": "^0.7.4",
    "express": "^4.17.2",
    "node-fetch": "2.6.1",
    "pinia": "^2.0.6",
    "vite-ssr": "^0.14.3",
    "vue": "^3.2.25",
    "vue-router": "^4.0.12"
  },
  "devDependencies": {
    "@types/express": "^4.17.13",
    "@typescript-eslint/eslint-plugin": "^5.7.0",
    "@typescript-eslint/parser": "^5.7.0",
    "@vitejs/plugin-vue": "^2.0.0",
    "autoprefixer": "^10.4.0",
    "eslint": "^8.4.1",
    "eslint-config-airbnb-base": "^15.0.0",
    "eslint-config-prettier": "^8.3.0",
    "eslint-import-resolver-typescript": "^2.5.0",
    "eslint-plugin-import": "^2.25.3",
    "eslint-plugin-prettier": "^4.0.0",
    "eslint-plugin-vue": "^8.2.0",
    "postcss": "^8.4.5",
    "prettier": "^2.5.1",
    "sass": "^1.45.0",
    "tailwindcss": "^3.0.6",
    "typescript": "^4.4.4",
    "vite": "^2.7.2",
    "vite-plugin-pages": "^0.19.4",
    "vue-tsc": "^0.29.8"
  }
}

````

vite.config.ts
````
/* eslint-disable import/no-extraneous-dependencies */
import { defineConfig } from 'vite';
import Vue from '@vitejs/plugin-vue';
import Pages from 'vite-plugin-pages';
import SSR from 'vite-ssr/plugin';

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [
    // See more at: https://v3.vuejs.org/guide/introduction.html/
    Vue(),
    // See more at: https://github.com/frandiox/vite-ssr/
    SSR(),
    // See more at: https://github.com/hannoeru/vite-plugin-pages/
    Pages({
      dirs: [
        { dir: 'src/routes/public/routes', baseRoute: '' },
        { dir: 'src/routes/admin/routes', baseRoute: '/admin' },
      ],
    }),
  ],
});

````

My routes folder looks like this.
![imagen](https://user-images.githubusercontent.com/47125485/146633263-b2bdcb67-62d7-4780-8f9b-39786236bd7d.png)

The thing is my whole project only detects the [...all].vue from public and the admin one is doing nothing.

Expected routes mapping:

/* -> 404 // then go to the main page
/admin/* -> 404 // then go to the admin main page.
...

Am I missing something like this?
![imagen](https://user-images.githubusercontent.com/47125485/146633410-bbfb4738-4e5a-45b6-8498-5c033842166f.png)
![imagen](https://user-images.githubusercontent.com/47125485/146633431-55c31441-af42-4a4d-aa22-a067c6ffccb8.png)

I'll be so gratefull if you guys can help me. Thanks!

Also If you need the code, pease ask for it and I'll make it public.

**Comment by TakenHiduk** [comment]
Hello @hannoeru It's me again, I think sometithing is broken now.

This is my folder structure
![imagen](https://user-images.githubusercontent.com/47125485
/146753249-61edbaff-0767-4933-91d6-c889ac61a1df.png)

Now I can get all the 404 as I espected, but... The index.vue is not loading. 

Here are some examples:
This should be "Hello from admin"
![imagen](https://user-images.githubusercontent.com/47125485/146753384-3378040d-743a-4733-9fa1-ec559ec0af4e.png)
But got this:
![imagen](https://user-images.githubusercontent.com/47125485/146753451-56aec4f8-d52f-4953-aab4-bdcd1257823f.png)

Also you can se in the url.

The same for /docs
![imagen](https://user-images.githubusercontent.com/47125485/146753523-b39e01cd-ee07-4352-943f-bd48d1611394.png)

It's something broken or just me? Also great work with this project, keep working on in. I love it.

**Comment by hannoeru** [comment]
Can you provide a repo with the code required to reproduce this issue?

**Comment by TakenHiduk** [comment]
Sure thing.

**Instalation**
```
yarn 
yarn dev
```

https://github.com/TakenHiduk/salepush.git

I'll make it private in some days. Feel free to use it as a template mean while.

**Comment by hannoeru** [comment]
There is a lot of priority issue so we only support one cache all route for now, but maybe we can add support for multiple cache all route in the feature.

**Comment by hannoeru** [comment]
This feature is now supported on beta channel, you can install it with `npm i vite-plugin-pages@beta -D`, please try it out and give me some feedback, thanks!

**Comment by Soya-xy** [comment]
> vite-plugin-pages@beta

Thanks   Let me try

**Comment by Soya-xy** [comment]
Thank you. It works normally, but I have a question after I try
When the generated path is `/child/map/: all (. *)`, it is not clear what the purpose is to include `(.*)` complete matching `/child/map/:all(.*)` Can be replaced with `/child/map/:all*`, matching `/child/map`

**Comment by hannoeru** [comment]
If we use `/path/:all*` it will match `/path` (the top got match first), which is not current as we are support `index` file.

<img width="1283" alt="スクリーンショット 2022-01-06 11 55 21" src="https://user-images.githubusercontent.com/16288797/148320937-81a7ba2b-abdf-4f89-bd10-dad0fff83f44.png">

**Comment by Soya-xy** [comment]
> 如果我们使用`/path/:all*`它将匹配`/path`（顶部首先匹配），这不是最新的，因为我们是支持`index`文件。
> 
> https://paths.esm.dev/?p=AAMeJRyAwBACXALcgOZQbQEqRkBaDgGoHUACBuJQcxRNxQ_4gGABpmQ5AAA.&t=/path#
> 
> <img alt="スクリーンショット 2022-01-06 11 55 21" width="1283" src="https://user-images.githubusercontent.com/16288797/148320937-81a7ba2b-abdf-4f89-bd10-dad0fff83f44.png">

oh,i see

**Comment by hannoeru** [comment]
Released v0.20.0
